server/log_rpc_server.go

```diff
@@ -506,31 +506,35 @@ func (t *TrillianLogRPCServer) GetEntryAndProof(ctx context.Context, req *trilli
 	if err != nil {
 		return nil, err
 	}
+	r := &trillian.GetEntryAndProofResponse{SignedLogRoot: &root}
 
-	proof, err := getInclusionProofForLeafIndex(ctx, tx, hasher, req.TreeSize, req.LeafIndex, root.TreeSize)
-	if err != nil {
-		return nil, err
-	}
+	// FIXME: should we return a proof to the available tree size if req.LeafIndex < root.TreeSize?
```

Comment:
No, I think that's best covered by the `SuperDoAllTheThings` RPC (that admittedly) doesn't yet exist); the client may well want the exact tree-size they asked for because they have a subset of particular STHs they want to use (and have separately confirmed to be consistent).